Bmx.Core.Test/State/MockAuthenticateState.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Bmx.Core.State;
+
+namespace Bmx.Core.Test.State {
+	public class MockAuthenticateState : IAuthenticateState {
```

Comment:
Curious: why did you choose a custom implementation for testing instead of mocking via `moq`?

Reply:
Good point, TBH I haven't really used Moq or NUnit much at all prior to this, investigating this.